app/services/vot/parser.rb

```diff
@@ -0,0 +1,96 @@
+module Vot
+  class Parser
+    class ParseException < StandardError; end
+    Result = Data.define(
+      :component_values,
+      :aal2?,
+      :phishing_resistant?,
+      :hspd12?,
+      :identity_proofing?,
+      :biometric_comparison?,
+      :ialmax?,
+    )
+
+    attr_reader :vector_of_trust
+
+    def initialize(vector_of_trust)
+      @vector_of_trust = vector_of_trust
+    end
+
+    def parse
+      initial_components = map_initial_vector_of_trust_components_to_component_values
+      expand_components_with_initial_components(initial_components)
+    end
+
+    def parse_acr
+      initial_components = map_initial_acr_values_to_component_values
+      expand_components_with_initial_components(initial_components)
+    end
```

Comment:
I think it might be clearer to use if we had the same `parse` method but different constructors? Since there are separate methods for parsing, it would be OK for them to use separate ivars. Or we could just pass in the string (and inline the error throwing methods)

```suggestion
    def parse
      initial_components = if vtr.present?
        map_initial_vector_of_trust_components_to_component_values
      elsif acr.present?
        map_initial_acr_values_to_component_values
      end
      expand_components_with_initial_components(initial_components)
    end
```

and then we have the constructor do like:
```ruby
def initialize(vtr:, acr:)
  @vtr = vtr
  @acr = acr
end
```